MIP/mip-61/README.md

```diff
@@ -0,0 +1,294 @@
+
+**2. BOOTSTRAPPING:** The relayer initiate the event polling at some point in the past. This point is either the genesis, some configurable source block height, or some checkpoint (possibly set on-chain)
+
+### Assumptions on the Bridge Protocol
+
+**Nonce**:
+We assume the Native Bridge protocol MUST implement the assignment of an incrementing `nonce` on the source chain. Since there are two directions there MUST be two types of counters (`nonces`) - one for each transfer direction. The `nonce` is used to order the transfers.
+
+**Initiate Transfer**:
+The `initiate_transfer` function on the source chain is called by the user. It emits an event `initiate_transfer` with the following properties:
+
+```javascript
+{
+    nonce: Number,          // Unique incrementing value for ordering transfers
+    recipient: String,  // Address of the recipient on the target chain
+    initiator: String,   // Address of the initiator on the source chain
+    amount: Number          // Amount being transferred
+}
+```
+
+This structure is used to record the transfer on the source chain.
+
+**Complete Transfer**:
+The `complete_transfer` function on the target chain is called by the relayer. It emits an event `complete_transfer` with the following properties:
+
+```javascript
+{
+    nonce: Number,         // Matching nonce from the source chain
+    transferUID: String,   // Unique identifier for the transfer
+    recipient: String,     // Address of the recipient
+    initiator: String,      // Address of the initiator
+    amount: Number         // Amount that was transferred
+}
+```
+
+This structure is used to record the transfer on the target chain.
+
+### Relayer Structures
+
+
+Locally the relayer stores the following structure:
+
+```javascript
+// Relayer Local Structure
+{
+    nonces: {                          // Object keyed by nonce (integer)
+        [nonce: Number]: {             // Nonce as the integer key
+            blockHeight: Number,       // Height of the block containing the nonce
+            status: String,            // Status of the event
+            eventUID: String           // Unique identifier for the event
+        }
+    }
+}
+```
+
+The relayer differentiates three status types:
+
+- `transfer_initiated`: the event has been received but not processed and the `complete_transfer` transaction has not been sent to the target chain.
+- `transfer_pending`: the `complete_transfer` transaction has been sent to the target chain but is not yet finalized.
+- `transfer_completed`: we received the (finalized) complete event.
+
+**Optimization**: There's another substate for pending when the `complete_transfer` transaction failed. In this case we should retry earlier than if the `complete_transfer` transaction was successful. We MAY differentiate the state `transfer_pending_executed` and `transfer_pending_failed`.
+
+### EVENT_PROCESSING algorithm
+
+For each `initiate_transfer` event do the following:
```

Comment:
We should monitor the `complete_transfer` event too. The goal of this event monitoring is to change the nonce state from pending to completed and process the `completed_block_height` and `completed_nonce_height`.
The other point is if the nonce doesn't exist in memory, it is added but the complete_transfer tx is not sent.

Reply:
ok. i added a section for the complete_transfer events

Reply:
My only remark is the graphic MIP/mip-61/bootstrap.png should be changed because bootstrap processes `initiate` and `complete` transfer events and in the graphic, the `iniitate_transfer` event is indicated as the only event. It should be replaced by `event` that is more general.

Reply:
I think the figure is correct. it says in the note above of it that a bootstrapping relayer 
1. ) starts BOOTSTRAP algo (which starts from an old block height or nonce
2.) starts in parrallel the CONTINUOUS_BLOCK_PROCESSING algo, which takes care of new `complete_transfer` events

Reply:
I'm saying that the bootstrap algo process initiate and complete event and in the diagram after `is end block` the `no` link  process only `initiate event,` the `complete event `should be added.

Reply:
Ok it's fine, I resolve.